app/src/main/java/com/github/sdpcoachme/groupevent/GroupEventDetailsActivity.kt

```diff
@@ -122,6 +125,12 @@ class GroupEventDetailsActivity : ComponentActivity() {
         // do not allow the activity to launch without an id. (double bang)
         val groupEventId = intent.getStringExtra(GROUP_EVENT_ID_KEY)!!
 
+        // todo move this into caching store
+        val locationProvider = (application as CoachMeApplication).locationProvider
+        locationProvider.updateContext(this, CompletableFuture.completedFuture(null))
+        val userLatLng = locationProvider.getLastLocation().value?: FusedLocationProvider.CAMPUS
+        val weatherState = store.getWeatherForecast(userLatLng).get()
```

Comment:
This could be handled like the other futures and added to the if not null to not have to do a get here. Please correct me if I'm wrong, I'm not sure about, but I think it would work out.

Reply:
Yes, I've implemented it this way before. But to be consistent with the `WeatherView` implementation that takes a `MutableState` as a parameter, I would have needed to use `var = remember { ... }` syntax and reassign a new MutableState inside the `LaunchedEffect` instead of changing its value. And as `getWeatherForecast` returns a synchronously completed future, I've decided to leave it as it is. Thanks for the comment and let me know what you think.